stripe/src/main/java/com/stripe/android/payments/PaymentFlowFailureMessageFactory.kt

```diff
@@ -0,0 +1,57 @@
+package com.stripe.android.payments
+
+import android.content.Context
+import com.stripe.android.R
+import com.stripe.android.StripeIntentResult
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.SetupIntent
+import com.stripe.android.model.StripeIntent
+
+internal class PaymentFlowFailureMessageFactory(
+    private val context: Context
+) {
+    fun create(
+        intent: StripeIntent,
+        @StripeIntentResult.Outcome outcome: Int
+    ): String? {
+        return when {
+            intent.status == StripeIntent.Status.RequiresPaymentMethod -> {
+                when (intent) {
+                    is PaymentIntent -> {
+                        when {
+                            intent.lastPaymentError?.code == PaymentIntent.Error.CODE_AUTHENTICATION_ERROR -> {
+                                context.resources.getString(R.string.stripe_failure_reason_authentication)
```

Comment:
🔭 👀  

(I see that it was like this but it's kind of a lot. Might be worth breaking this down into a few functions.)